app/lib/annotation_viz_service.rb

```diff
@@ -139,34 +130,27 @@ def self.available_annotations(study, cluster: nil, current_user: nil, annotatio
       end
     end
     annotations.concat(cluster_annots)
-    # show 'invalid' annotations (e.g. cannot be visualized) as a courtesy, but don't allow selection
-    invalid_annots = []
-    study.cell_metadata.each do |meta|
-      unless meta.can_visualize?
-        invalid_annots << {
-          name: meta.name,
-          type: meta.annotation_type,
-          values: meta.values,
-          scope: 'invalid'
-        }
-      end
-    end
-    if cluster.present?
-      cluster.cell_annotations.each do |cell_annot|
-        unless cluster.can_visualize_cell_annotation?(cell_annot)
-          invalid_annots << {
-            name: cell_annot['name'],
-            type: cell_annot['annotation_type'],
-            values: cell_annot['values'],
-            scope: 'invalid'
-          }
-        end
-      end
-    end
-    annotations.concat(invalid_annots)
     annotations
   end
 
+  # helper method to efficiently list out metadata annotations classed as valid/invalid for visualization
+  def self.available_metadata_annotations(study, annotation_type: nil)
+    # get all the metadata in a single query
+    all_metadata = study.cell_metadata.to_a
+    all_names = all_metadata.map(&:name)
+    all_metadata.map do |annot|
+      # viewable if the type is numeric or there's no corresponding label and it's within the range of visualization values
+      is_viewable = annot.annotation_type == 'numeric' ||
```

Comment:
This can be massively simplified.  It does include another DB call, but doing `exists?` does not actually load any data, and is incredibly fast (sub-100ms)
```suggestion
      is_viewable = annot.can_visualize?
```

Reply:
it's fast, but it would still require one DB call per metadata column, so for the typical convention-compliant study, that's 12ish DB calls, and then those twelve would repeat for the `explore` API call.  So we're saving ~24 DB calls.  At ~0.05sec per call (what I see locally for DB call time), that's a whole second